tests/chainerx_tests/unit_tests/routines_tests/test_creation.py

```diff
@@ -1173,3 +1176,37 @@ def test_copy(xp, shape, dtype, device, is_module):
         return xp.copy(a)
     else:
         return a.copy()
+
+
+@op_utils.op_test(['native:0', 'cuda:0'])
+@chainer.testing.parameterize_pytest('indexing', ['xy', 'ij'])
+@chainer.testing.parameterize_pytest('input_arrs', [2, 3, 4, 5, 6])
+class TestMeshgrid(op_utils.NumpyOpTest):
+
+    check_numpy_strides_compliance = False
+
+    def setup(self, dtype):
+        # TODO(kshitij12345) : Remove when #6621 is in.
+        if numpy.dtype(dtype).kind != 'f':
+            self.skip_backward_test = True
+            self.skip_double_backward_test = True
+        self.dtype = dtype
+
+        if dtype == 'float16':
+            self.check_backward_options.update({'rtol': 1e-2, 'atol': 1e-2})
+
+    def generate_inputs(self):
+        arrs = ()
+        for _ in range(self.input_arrs):
+            arrs += (numpy.linspace(randint(-10, 0), randint(1, 10),
+                                    randint(3, 7)).astype(self.dtype),)
+
+        return arrs
+
+    def forward_xp(self, inputs, xp):
+        if xp is numpy:
+            outs = xp.meshgrid(*inputs, indexing=self.indexing)
+        elif xp is chainerx:
+            outs = xp.meshgrid(list(inputs), indexing=self.indexing)
```

Comment:
Can you alight ChainerX's meshgrid binding so that this conditioning is not needed? I think you can use `py::args` instead of `py::sequence` but I have not tried.

Reply:
Done.